conbench/job.py

```diff
@@ -100,28 +100,39 @@ def _fetch_and_cache_most_recent_results(n=0.08 * 10**6) -> None:
     )
     t0 = time.monotonic()
 
+    # Note(JP): process query result rows in a streaming-like fashion in
+    # smaller chunks to keep peak memory usage in check. Also see
+    # https://docs.sqlalchemy.org/en/20/core/connections.html#using-server-side-cursors-a-k-a-stream-results
+    # https://docs.sqlalchemy.org/en/20/orm/queryguide/api.html#fetching-large-result-sets-with-yield-per
     # Also fetch hardware from associated Run, so that result.run is in cache,
     # too, and so that result.run.hardware is a quick lookup.
-    # Note that N here determines peak usage of memory.
     query_statement = (
         sqlalchemy.select(BenchmarkResult)
         .join(Run)
         .order_by(BenchmarkResult.timestamp.desc())
         .limit(n)
-    )
-    results = Session.scalars(query_statement).all()
+    ).execution_options(yield_per=2000)
```

Comment:
I chose the chunk size on the one hand arbitrarily, on the other hand of course this is a number that absolutely tries to keep the iteration overhead minimal while at the same time significantly changing N. That is:
```
1 << 2000 << 60000
```